diff --git a/WalletWasabi/Transactions/ITxStoreOperation.cs b/WalletWasabi/Transactions/ITxStoreOperation.cs
index 5c4702fd..bd5e8d32 100644
--- a/WalletWasabi/Transactions/ITxStoreOperation.cs
+++ b/WalletWasabi/Transactions/ITxStoreOperation.cs
@@ -6,6 +6,6 @@ namespace WalletWasabi.Transactions
 {
 	public interface ITxStoreOperation
 	{
-		public bool IsEmpty { get; }
+		bool IsEmpty { get; }
 	}
 }
