diff --git a/WalletWasabi.Backend/Program.cs b/WalletWasabi.Backend/Program.cs
index fc5cbe51..95789a6c 100644
--- a/WalletWasabi.Backend/Program.cs
+++ b/WalletWasabi.Backend/Program.cs
@@ -22,7 +22,7 @@ namespace WalletWasabi.Backend
 		{
 			try
 			{
-				var endPoint = "http://localhost:37127/";
+				var endPoint = "http://wasabi-backend:37127/";
 
 				using var host = Host.CreateDefaultBuilder(args)
 					.ConfigureWebHostDefaults(webBuilder => webBuilder
