From de75327d5608d190650bba149582a322d09853d1 Mon Sep 17 00:00:00 2001
From: SKP <jash@schulterklopfer-productions.de>
Date: Mon, 9 Sep 2019 19:28:59 +0200
Subject: [PATCH] Regtest now uses bitcoin core node from settings

---
 WalletWasabi.Gui/Global.cs | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/WalletWasabi.Gui/Global.cs b/WalletWasabi.Gui/Global.cs
index 319b7d17..6704fc83 100644
--- a/WalletWasabi.Gui/Global.cs
+++ b/WalletWasabi.Gui/Global.cs
@@ -244,10 +244,13 @@ namespace WalletWasabi.Gui
 				Nodes = new NodesGroup(Network, requirements: Constants.NodeRequirements);
 				try
 				{
-					Node node = await Node.ConnectAsync(Network.RegTest, new IPEndPoint(IPAddress.Loopback, 18444));
+					EndPoint bitcoinCoreEndpoint = Config.GetBitcoinP2pEndPoint();
+
+					Node node = await Node.ConnectAsync(Network.RegTest, bitcoinCoreEndpoint);
+
 					Nodes.ConnectedNodes.Add(node);
 
-					RegTestMempoolServingNode = await Node.ConnectAsync(Network.RegTest, new IPEndPoint(IPAddress.Loopback, 18444));
+					RegTestMempoolServingNode = await Node.ConnectAsync(Network.RegTest, bitcoinCoreEndpoint);
 
 					RegTestMempoolServingNode.Behaviors.Add(new MempoolBehavior(MempoolService));
 				}
-- 
2.21.0

