diff --git a/WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs b/WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs
index 7a39684e..7d8e40af 100644
--- a/WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs
+++ b/WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs
@@ -98,6 +98,7 @@ namespace WalletWasabi.Gui.Rpc
 			public BitcoinAddress sendto;
 			public Money amount;
 			public string label;
+			public bool subtractFee;
 		}
 
 		[JsonRpcMethod("send")]
@@ -105,7 +106,7 @@ namespace WalletWasabi.Gui.Rpc
 		{
 			AssertWalletIsLoaded();
 			var sync = _global.Synchronizer;
-			var payment = new PaymentIntent( payments.Select(p=> new DestinationRequest(p.sendto.ScriptPubKey, MoneyRequest.Create(p.amount), new SmartLabel(p.label))));
+			var payment = new PaymentIntent(payments.Select((p,i)=> new DestinationRequest(p.sendto.ScriptPubKey, MoneyRequest.Create(p.amount, p.subtractFee), new SmartLabel(p.label))));
 			var feeStrategy = FeeStrategy.CreateFromConfirmationTarget(feeTarget);
 			var password = string.Empty;
 			var result = _global.WalletService.BuildTransaction(
